ts/features/messages/hooks/useAttachmentDownload.tsx

```diff
@@ -42,17 +46,41 @@ export const useAttachmentDownload = (
     isDownloadingMessageAttachmentSelector(state, messageId, attachmentId)
   );
 
+  const attachmentCategory = attachment.category;
   const doNavigate = useCallback(() => {
     dispatch(clearRequestedAttachmentDownload());
-    navigation.navigate(MESSAGES_ROUTES.MESSAGES_NAVIGATOR, {
-      screen: MESSAGES_ROUTES.MESSAGE_DETAIL_ATTACHMENT,
-      params: {
-        messageId,
-        serviceId,
-        attachmentId
-      }
-    });
-  }, [attachmentId, dispatch, messageId, navigation, serviceId]);
+    onPreNavigate?.();
+    if (isPN) {
+      trackPNAttachmentOpening(attachmentCategory);
+      NavigationService.navigate(MESSAGES_ROUTES.MESSAGES_NAVIGATOR, {
+        screen: PN_ROUTES.MAIN,
+        params: {
+          screen: PN_ROUTES.MESSAGE_ATTACHMENT,
+          params: {
+            attachmentId,
+            messageId
+          }
+        }
+      });
```

Comment:
why not use `useIONavigation`?  AFAIK the `NavigationService` was deprecated or at least heavily discouraged

Reply:
This hook is also references inside a bottom sheet, where the navigation context of the `useNavigation` hook is not available. Hence the need of using the `NavigationService` (which will not be deprecated due to this use case)